packages/rsbuild-plugin/src/cli/index.ts

```diff
@@ -0,0 +1,155 @@
+import { parseOptions } from '@module-federation/enhanced';
+import { ModuleFederationPlugin } from '@module-federation/enhanced/rspack';
+
+import { isRegExp, isRequiredVersion } from '../utils/index';
+import pkgJson from '../../package.json';
+
+import type {
+  moduleFederationPlugin,
+  sharePlugin,
+} from '@module-federation/sdk';
+import type { RsbuildPlugin, EnvironmentConfig } from '@rsbuild/core';
+
+type ModuleFederationOptions =
+  moduleFederationPlugin.ModuleFederationPluginOptions;
+
+const PLUGIN_MODULE_FEDERATION_NAME = 'rsbuild:module-federation-enhanced';
+export const pluginModuleFederation = (
+  moduleFederationOptions: ModuleFederationOptions,
+): RsbuildPlugin => ({
+  name: PLUGIN_MODULE_FEDERATION_NAME,
+  setup: (api) => {
+    const sharedOptions: [string, sharePlugin.SharedConfig][] = parseOptions(
+      moduleFederationOptions.shared || [],
+      (item, key) => {
+        if (typeof item !== 'string')
+          throw new Error('Unexpected array in shared');
+        const config: sharePlugin.SharedConfig =
+          item === key || !isRequiredVersion(item)
+            ? {
+                import: item,
+              }
+            : {
+                import: key,
+                requiredVersion: item,
+              };
+        return config;
+      },
+      (item) => item,
+    );
+    // shared[0] is the shared name
+    const shared = sharedOptions.map((shared) =>
+      shared[0].endsWith('/') ? shared[0].slice(0, -1) : shared[0],
+    );
+
+    api.modifyRsbuildConfig((config) => {
+      if (
```

Comment:
Keep the comments: https://github.com/web-infra-dev/rsbuild/blob/main/packages/core/src/plugins/moduleFederation.ts#L102-L103

```suggestion
      // If this is a provider app, Rsbuild should send the ws request to the provider's dev server.
      // This allows the provider to do HMR when the provider module is loaded in the consumer's page.
      if (
```